src/main/java/bob/MainWindow.java

```diff
@@ -0,0 +1,109 @@
+package bob;
+
+import javafx.animation.PauseTransition;
+import javafx.application.Platform;
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.VBox;
+import javafx.util.Duration;
+
+/**
+ * Controller for the main GUI of the Bob chatbot application.
+ * Manages user interactions and updates the dialog container with messages.
+ */
+public class MainWindow extends AnchorPane {
+    @FXML
+    private ScrollPane scrollPane;
+    @FXML
+    private VBox dialogContainer;
+    @FXML
+    private TextField userInput;
+    @FXML
+    private Button sendButton;
+
+    private Bob bob;
+    private Image userImage = new Image(this.getClass().getResourceAsStream("/images/DaUser.png"));
+    private Image bobImage = new Image(this.getClass().getResourceAsStream("/images/DaBob.png"));
+
+    /**
+     * Initializes the controller. Binds the scroll pane's vertical value property to the dialog container's height
+     * to keep the scroll pane scrolled to the bottom.
+     */
+    @FXML
+    public void initialize() {
+        scrollPane.vvalueProperty().bind(dialogContainer.heightProperty());
+    }
+
+    /**
+     * Sets the Bob instance for this controller.
+     *
+     * @param b The Bob instance to be set.
+     */
+    public void setBob(Bob b) {
+        this.bob = b;
+    }
+
+    /**
+     * Displays the welcome message from Bob in the dialog container.
+     * This method should be called after setting the Bob instance to ensure proper initialization.
+     */
+    public void showWelcomeMessage() {
+        if (this.bob != null && this.bob.getUi() != null) {
+            displayMessage(this.bob.getUi().showWelcomeMessage("Bob"), bobImage);
+        } else {
+            System.err.println("Bob or Ui has not been initialized.");
+        }
+    }
+
+    /**
+     * Handles the user input from the text field. Displays the user input and the corresponding response from Bob in
+     * the dialog container. Clears the user input field after processing.
+     * If the user input is "bye", displays the goodbye message and closes the application after a short delay.
+     */
+    @FXML
+    private void handleUserInput() {
```

Comment:
I like how u made use of comments to describe the inner logic, but some comments could perhaps be omitted as it is trivial